src/plugins/world_control/CMakeLists.txt

```diff
@@ -1,9 +1,19 @@
-ign_gui_add_plugin(WorldControl
-  SOURCES
-    WorldControl.cc
-  QT_HEADERS
-    WorldControl.hh
-  TEST_SOURCES
-    WorldControl_TEST.cc
-)
-
+if (NOT MSVC)
```

Comment:
Just curious, why did you need to separate Windows here? We're already disabling the test on Windows in the test's body.

Reply:
Because Windows would still compile the testing framework and produce compile time errors.

Reply:
I see. It was compiling before. Not sure what causes the compiler error here, but I think it's ok to address this later.

Reply:
As a follow-up, yes,the tests themselves ignore windows. For example: https://github.com/ignitionrobotics/ign-gui/blob/ignition-gui6_6.0.0/src/plugins/world_control/WorldControl_TEST.cc#L66

However, the new helper test class that was added for checking events (https://github.com/ignitionrobotics/ign-gui/pull/306#issuecomment-959731077) has to be added as a QT header, or else there will be linking errors. So, it seemed like this new QT header is what was causing issues on windows, and @nkoenig made a patch in https://github.com/ignitionrobotics/ign-gui/pull/306#issuecomment-961378270 to prevent compiling the helper test class header on windows. I'm not sure if there's a better way to handle this or not, but hopefully that provides some more context/insights about this change.

Reply:
I tried a few different options to make the header compile on windows, and gave up.